firmware_v5/telelogger/config.md
# Arquivo de configurações:

## CSV Logging Configuration:
---

### SHOULD_OBD_LOG e SHOULD_GPS_LOG:

    Definem respectivamente se os dados do OBD e do GPS deverão ser salvos.

## Circular Buffer Configuration:
---

### BUFFER_SLOTS e BUFFER_LENGTH:

    BUFFER_SLOTS define a quantidade de slots de tamanho BUFFER_LENGTH bytes a serem armazenados no buffer.

### SERIALIZE_BUFFER_SIZE:

    Esta variável precisa de mais investigações.

## Configuration Definitions:
---

    Definições de configuração de dispositivo de rede, acelerômetro, armazenamento, protocolo e método de requisição.

## OBD-II configurations:
---

### ENABLE_OBD:

    Define se o OBD será utilizado.

### MAX_OBD_ERRORS:

    Quantidade de erros de OBD permitidos antes de entrar em standby.

## Networking Configurations:
---

### NET_DEVICE:

    Define o dispositivo de rede a ser utilizado. (NET_WIFI para utilizar Wifi e NET_SIM5360 para utilizar o 3G brasileiro).

### WIFI_SSID e WIFI_PASSWORD:

    Nome e senha da rede WiFi. Se NET_DEVICE não estiver definido como NET_WIFI, estes valores serão desconsiderados.

### CELL_APN:

    Endereço APN da operadora.

### SERVER_HOST e SERVER_PROTOCOL:

    Endereço do host e protocolo a ser utilizado no servidor.

### SIM_CARD_PIN:

    PIN do cartão SIM, caso necessário.

### SERVER_METHOD e SERVER_PATH:

    Método de requisição e caminho da api no servidor.

### SERVER_PORT:

    Porta utilizada pelo servidor.

### WIFI_MESH_ID e WIFI_MESH_CHANNEL:

    ID e canal da Wifi Mesh (a ser estudado).

### WIFI_AP_SSID e WIFI_AP_PASSWORD:

    ID e senha do Acess Point, se necessário.

### MAX_CONN_ERRORS_RECONNECT:

    Número máximo de erros de conexão antes de tentar reconectar.

### MAX_CONN_TIME:

    Não encontrei nenhuma referência a MAX_CONN_TIME no código.

### DATA_RECEIVING_TIMEOUT:

    Define o intervalo esperado para recebimento dos dados do servidor.

### SERVER_SYNC_INTERVAL:

    Define o tempo máximo esperado para sincronização com o servidor.

### STATIONARY_TIME_TABLE e SENDING_INTERVAL_TABLE:

    O intervalo para enviar dados ao servidor é definido pelo tempo em que o veículo se encontra estacionário.
> Por exemplo,
> Se definirmos STATIONARY_TIME_TABLE = {30, 60, 180} e SENDING_INTERVAL_TABLE = {0, 2000, 5000}, o intervalo de envio será 0 ms quando o tempo estacionário t < 30s, será de 2000 ms se 30 < t < 60 e 5000 se t > 60.

### DATASET_INTERVAL:

    Tempo mínimo de cada ciclo de processamento de dados (antes de inicializar a rede para enviá-los).
    Depois de processar os dados do OBD e do GPS, existe o delay de t ms, onde t = DATASET_INTERVAL - tempo decorrido no ciclo.

### PING_BACK_INTERVAL:

## Data storage configurations:
---

### STORAGE:

    Define o método de armazenamento de dados.

## MEMS sensors:
---

### ENABLE_ORIENTATION:

    Define se a orietação por acelerômetro será utilizada.

### MEMS_MODE:

    Define o modo do acelerômetro (a ser estudado).

## GPS:
---

### ENABLE_GPS:

    Define se o GPS será utilizado.

### GPS_SERIAL_BAUDRATE:

    Define a Baudrate utilizada pelo GPS.

### GPS_MOTION_TIMEOUT:

    Não encontrei uso para GPS_MOTION_TIMEOUT. A única referência a ela é num #if 0 que nunca acontece.

## Standby/wakeup:
---

### RESET_AFTER_WAKEUP:

    Define se o dispositivo deve reiniciar (1) ou não (0) depois de sair de Standby.

### MOTION_THRESHOLD:

    Limiar de aceleração definido em G's (usado para identificar se há movimento no carro a partir do acelerômetro)

### JUMPSTART_VOLTAGE:

    Limiar de tensão da bateria para sair do modo de Standby.

## Additional features:
---

### ENABLE_HTTPD:

### ENABLE_OLED:

### CONFIG_MODE_TIMEOUT:

### PIN_SENSOR1 35:

### PIN_SENSOR2 36:

### COOLING_DOWN_TEMP:

firmware_v5/telelogger/config.h
#ifndef CONFIG_H_INCLUDED
#define CONFIG_H_INCLUDED

/**************************************
* CSV Logging Configuration
**************************************/

#define SHOULD_OBD_LOG true
#define SHOULD_GPS_LOG false

/**************************************
* Circular Buffer Configuration
**************************************/
#define BUFFER_SLOTS 32 /* max number of buffer slots to be saved */
#define BUFFER_LENGTH 128 /* bytes per each saved slot */
#define SERIALIZE_BUFFER_SIZE 1024 /* bytes */

/**************************************
* Configuration Definitions
**************************************/
#define NET_WIFI 1
#define NET_SIM800 2
#define NET_SIM5360 3
#define NET_SIM7600 4
#define NET_WIFI_MESH 5
#define NET_SERIAL 6

#define MEMS_DISABLED 0
#define MEMS_ACC 1
#define MEMS_9DOF 2
#define MEMS_DMP 3

#define STORAGE_NONE 0
#define STORAGE_SPIFFS 1
#define STORAGE_SD 2

#define PROTOCOL_UDP 1
#define PROTOCOL_HTTPS 2

#define PROTOCOL_METHOD_GET 0
#define PROTOCOL_METHOD_POST 1

/**************************************
* OBD-II configurations
**************************************/
#ifndef ENABLE_OBD
#define ENABLE_OBD 1
#endif

// maximum consecutive OBD access errors before entering standby
#define MAX_OBD_ERRORS 3

/**************************************
* Networking configurations
**************************************/
#ifndef NET_DEVICE
// change the following line to change network device
#define NET_DEVICE NET_WIFI
// WiFi settings
#define WIFI_SSID "jordy"
#define WIFI_PASSWORD "roteador"
// cellular network settings
#define CELL_APN "tim.brasil.br"
// Freematics Hub server settings
#define SERVER_HOST "hub.freematics.com"
#define SERVER_PROTOCOL PROTOCOL_UDP
#endif 

// SIM card setting
#define SIM_CARD_PIN ""

// HTTPS settings
#define SERVER_METHOD PROTOCOL_METHOD_POST
#define SERVER_PATH "/hub/api"

#if !SERVER_PORT
#if SERVER_PROTOCOL == PROTOCOL_UDP
#define SERVER_PORT 8081
#elif SERVER_PROTOCOL == PROTOCOL_HTTPS
#define SERVER_PORT 443
#endif
#endif

// WiFi Mesh settings
#define WIFI_MESH_ID "123456"
#define WIFI_MESH_CHANNEL 13

// WiFi AP settings
#define WIFI_AP_SSID "TELELOGGER"
#define WIFI_AP_PASSWORD "PASSWORD"

// maximum consecutive communication errors before reconnecting
#define MAX_CONN_ERRORS_RECONNECT 3
// maximum allowed connecting time
#define MAX_CONN_TIME 10000 /* ms */
// data receiving timeout
#define DATA_RECEIVING_TIMEOUT 5000 /* ms */
// expected maximum server sync signal interval
#define SERVER_SYNC_INTERVAL 120 /* seconds, 0 to disable */
// data interval configurations
#define STATIONARY_TIME_TABLE {30, 60, 180} /* seconds */ // The interval to send data to the server will be defined by the stationary time. If the car is
#define SENDING_INTERVAL_TABLE {0, 2000, 5000} /* ms */   // stationary for t < 30s, the sending interval would be 0. If 30 < t < 60, it will be 2000, and so on.
#define DATASET_INTERVAL 200 // Interval for each dataset to be saved (it actually calls a delay function every DATASET_INTERVAL seconds)
#define PING_BACK_INTERVAL 900 /* seconds */

/**************************************
* Data storage configurations
**************************************/
#ifndef STORAGE
// change the following line to change storage type
#define STORAGE STORAGE_SD
#endif

/**************************************
* MEMS sensors
**************************************/
#define ENABLE_ORIENTATION 0
#ifndef MEMS_MODE
#define MEMS_MODE MEMS_9DOF
#endif

/**************************************
* GPS
**************************************/
#ifndef ENABLE_GPS
// change the following line to enable (1)/disable (0) GPS
#define ENABLE_GPS 1
#endif
#define GPS_SERIAL_BAUDRATE 115200L
#define GPS_MOTION_TIMEOUT 180 /* seconds */

/**************************************
* Standby/wakeup
**************************************/
#define RESET_AFTER_WAKEUP 1
// motion threshold for waking up
#define MOTION_THRESHOLD 0.3f /* moving vehicle motion threshold in G */
// engine jumpstart voltage
#define JUMPSTART_VOLTAGE 14 /* V */

/**************************************
* Additional features
**************************************/
#define ENABLE_HTTPD 0
#define ENABLE_OLED 0
#define CONFIG_MODE_TIMEOUT 0

#define PIN_SENSOR1 35
#define PIN_SENSOR2 36

#define COOLING_DOWN_TEMP 65 /* celsius degrees */

#endif // CONFIG_H_INCLUDED
